File: src/test_programs/F03/test_init.f90
program test_init
    use iso_c_binding, only: c_char
    use ommp_interface

    implicit none
    character(kind=c_char, len=120), dimension(2) :: args
    integer :: narg
    type(ommp_system), pointer :: my_system
  
    narg = command_argument_count()
    if (narg /= 2 .and. narg /= 1) then
        write(6, *) "Syntax expected "
        write(6, *) "   $ test_init.exe <INPUT FILE> [<OUTPUT FILE>]"
    else 
        call get_command_argument(1, args(1))
        if(narg == 2) &
            call get_command_argument(2, args(2))
       
        call ommp_set_verbose(OMMP_VERBOSE_DEBUG)
        call ommp_init_mmp(my_system, trim(args(1)))
        if(narg == 2) then
             call ommp_print_summary_to_file(my_system, args(2))
        else
            call ommp_print_summary(my_system)
        end if
        
        call ommp_terminate(my_system)
    end if
end program test_init

File: src/test_programs/test_init.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "openmmpol.h"

int main(int argc, char **argv){
    if(argc != 3 && argc != 2){
        printf("Syntax expected\n");
        printf("    $ test_init.exe <INPUT FILE> [<OUTPUT FILE>]\n");
        return 0;
    }
    
    char infile[120], outfile[120];
    strcpy(infile, argv[1]);
    
    if(argc == 3)
        strcpy(outfile, argv[2]);
        
    OMMP_SYSTEM_PRT my_system = ommp_init_mmp(infile);
    ommp_set_verbose(OMMP_VERBOSE_DEBUG);
    if(argc == 3)
        ommp_print_summary_to_file(my_system, outfile);
    else
        ommp_print_summary(my_system);
    
    ommp_terminate(my_system);
    
    return 0;
}
